content/flux/v0.x/join-data/full-outer.md

```diff
@@ -0,0 +1,254 @@
+| r1  | <span style="color:#d30971">▲</span> | <span style="color:#d30971">▲</span> |
+| r3  | <span style="color:#d30971">▲</span> | <span style="color:#d30971">▲</span> |
+| r4  | <span style="color:#d30971">▲</span> | <span style="color:#d30971">▲</span> |
+{{% /flex-content %}}
+{{% flex-content "third" %}}
+#### Full outer join result
+
+|     |                                      |                                      |                                      |                                      |
+| :-- | :----------------------------------- | :----------------------------------- | :----------------------------------- | :----------------------------------- |
+| r1  | <span style="color:#9b2aff">●</span> | <span style="color:#9b2aff">●</span> | <span style="color:#d30971">▲</span> | <span style="color:#d30971">▲</span> |
+| r2  | <span style="color:#9b2aff">●</span> | <span style="color:#9b2aff">●</span> |                                      |                                      |
+| r3  |                                      |                                      | <span style="color:#d30971">▲</span> | <span style="color:#d30971">▲</span> |
+| r4  |                                      |                                      | <span style="color:#d30971">▲</span> | <span style="color:#d30971">▲</span> |
+{{% /flex-content %}}
+{{< /flex >}}
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+## Prepare your data
+
+Ensure the data streams you want to join adhere to the
+[join input data requirements](/flux/v0.x/join-data/#data-requirements).
+
+## Use join.full to join your data
+
+1. Import the `join` package.
+2. Define two streams of tables to join.
+
+    One stream of table represents the left side of the join.
+    The other stream of table represents the right side of the join.
+
+3. Use `join.full()` to join the two streams together.
+    Provide the following parameters:
+
+    - `left`: stream of data representing the left side of the join
+    - `right`: stream of data representing the right side of the join
+    - `on`: [join predicate](/flux/v0.x/join-data/#join-predicate-function-on)
+    - `as`: [join output function](/flux/v0.x/join-data/#join-output-function-as)
+      that returns a record with values from each input stream
+
+      {{% note %}}
+#### Account for missing, non-group-key values
+
+With the output function used in the `on` parameter with `join.full()`
+it’s possible for either `1` or `r` to use the default record (a record with
+group key columns populated but other columns are _null_).
+`join.full()` guarantees that `l` and `r` will never both use default records at
+the same time and will have the non-null values needed for the output record
+
+For each non-group-key column you want to ensure has a non-null value in the output,
+use a conditional statement to check for the existence of a value in the `l` or `r`
+record and return the value that exists:
```

Comment:
Can we make it clear that this check is only necessary for full outer joins? Left, right, and inner joins don't need this check.

Reply:
Nevermind, I just realized this doc is specifically for `join.full()`. Disregard.